src/factor_graph.h

```diff
@@ -101,17 +101,21 @@ class CompactFactorGraph {
  public:
   FactorGraphDescriptor size;
 
-  std::unique_ptr<Variable[]> variables;
-  std::unique_ptr<Factor[]> factors;
-
   // For each edge, we store the factor, weight id, factor id, and the variable,
   // in the same index of seperate arrays. The edges are ordered so that the
   // edges for a variable is in a continuous region (sequentially).
   // This allows us to access factors given variables, and access variables
   // given factors faster.
+
+  // Variables and edges indexed by variable (sort by <variable, factor>)
+  // Note that wids are put in a dedicated array (instead of in CompactFactor)
+  // because we'd like to compact memory write footprint (????????)
```

Comment:
@netj @zhangce @SenWu is this speculation about the existence of `compact_factors_weightids` correct?